src/lightning/fabric/utilities/cloud_io.py

```diff
@@ -70,3 +72,56 @@ def _atomic_save(checkpoint: Dict[str, Any], filepath: Union[str, Path]) -> None
     torch.save(checkpoint, bytesbuffer)
     with fsspec.open(filepath, "wb") as f:
         f.write(bytesbuffer.getvalue())
+
+
+@lru_cache(maxsize=None)
+def _lazy_import(module_name: str) -> Optional[ModuleType]:
```

Comment:
can we rather use https://github.com/Lightning-AI/utilities/blob/e7cb7cbbd3473ea13701ce12682c1d696681e89d/src/lightning_utilities/core/imports.py#L209

Reply:
Great suggestion, I wasn't aware of this utilities package! I slightly reorganized the checks to use the `lazy_import` function.